src/components/date-picker/calendar.tsx

```diff
@@ -201,19 +227,41 @@ export class Calendar extends React.Component<CalendarProps, CalendarState> {
         monthNames.forEach(month => {
             monthArray.push(
                 <span
-                    className="calendarItem monthName"
+                    className="monthName"
                     onMouseDown={this.onSelectMonth}
                     key={`MONTH_${month.toUpperCase()}`}
                     data-automation-id={`MONTH_${month.toUpperCase()}`}
                 >
                     {month}
                 </span>
-                );
+            );
         });
 
         return monthArray;
     }
 
+    @computed
+    private get yearArray(): JSX.Element[] {
+        const yearArray: JSX.Element[] = [];
+
+        for (let year = this.state.viewDate.getFullYear() - 5; year <= this.state.viewDate.getFullYear() + 5; year ++) {
+            if (year === this.props.value.getFullYear()) { continue; }
```

Comment:
Just a suggestion: maybe show the currently selected year and make it highlighted?
1) It would solve the problem with variable number of years in the popup.
2) Right now you need to perform two different actions depending on which year you want to select. You either need to click on the header or on the item in the list.